pluto-message-ingestion/pluto-message-processor.js

```diff
@@ -0,0 +1,95 @@
+class PlutoMessageProcessor {
+  constructor({ hostname, hmacRequest, logger }) {
+    this.hostname = hostname;
+    this.hmacRequest = hmacRequest;
+    this.logger = logger;
+  }
+
+  process(message) {
+    switch (message.type) {
+      case 'project-created': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._createProject(project)
+            .then(resp => {
+              this.logger.log('successfully created project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              this.logger.error('failed to create project', {
+                status: err.status,
+                response: err.response,
+                project: project
+              });
+              reject(err);
+            });
+        });
+      }
+      case 'project-updated': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._updateProject(project)
+            .then(resp => {
+              this.logger.log('successfully updated project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              const logDetail = {
+                status: err.status,
+                response: err.response,
+                project: project
+              };
+
+              if (err.status === 404) {
```

Comment:
@akash1810 yes I am afraid projects can span many months.  I would suggest failing back to the "project-created" behaviour if an update message is received for a project that does not exist yet; the only alternative i can think of is a one-time hit to ingest all current projects into the system before sending it live.